bridgepoint/prebuild.py

```diff
@@ -1609,11 +1615,34 @@ def find_symbol(self, node, name):
             v_var = one(v_int).V_VAR[814]()
         
         return v_var
-    
-    @property
+
+    def get_event_name(self, event):
```

Comment:
This method is difficult to read and understand.

Could you flatten the conditions a bit, and add comment on each case?
I tried my best to write a sugestion, but I do not have access to the metamodel at the moment.
Here is a possibly incorrect example that points towards my intentions.


```
def get_event_name(sm_evt):
    sm_nlevt = one(sm_evt).SM_SEVT[525].SM_NLEVT[526]()
    sm_sgevt = one(sm_evt).SM_SEVT[525].SM_SGEVT[526]()
    sm_pect = one(sm_nlevt).SM_PEVT[527]()

    # non-local polymorphic event
    if sm_pect and sm_pect.SM_ID != sm_nlevt.SM_ID:
        return sm_evt.Drv_Lbl + ": " + sm_evt.Mning

    # local polymorphic event
    if sm_pect:
        return sm_evt.Mning + "::" + sm_pect.localClassName

    # orphan, what is this?
    if sm_nlevt:
        return event.Mning + "::Orphaned"

    # case 4
    if sm_sgevt:
        return sm_evt.Drv_Lbl

    # case 5
    return sm_evt.Drv_Lbl + ": " + sm_evt.Mning

```